adapters/alkimi/alkimi.go

```diff
@@ -0,0 +1,195 @@
+}
+
+// Builder builds a new instance of the Alkimi adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	endpointURL, err := url.Parse(config.Endpoint)
+	if err != nil || len(endpointURL.String()) == 0 {
+		return nil, fmt.Errorf("invalid endpoint: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: endpointURL.String(),
+	}
+	return bidder, nil
+}
+
+// MakeRequests creates Alkimi adapter requests
+func (adapter *adapter) MakeRequests(request *openrtb2.BidRequest, req *adapters.ExtraRequestInfo) (reqsBidder []*adapters.RequestData, errs []error) {
+	reqCopy := *request
+
+	updated, errs := updateImps(reqCopy)
+	if len(errs) > 0 || len(reqCopy.Imp) != len(updated) {
+		return nil, errs
+	}
+
+	reqCopy.Imp = updated
+	encoded, err := json.Marshal(reqCopy)
+	if err != nil {
+		errs = append(errs, err)
+	} else {
+		reqBidder := buildBidderRequest(adapter, encoded)
+		reqsBidder = append(reqsBidder, reqBidder)
+	}
+	return
+}
+
+func updateImps(bidRequest openrtb2.BidRequest) ([]openrtb2.Imp, []error) {
+	var errs []error
+
+	updatedImps := make([]openrtb2.Imp, 0, len(bidRequest.Imp))
+	for _, imp := range bidRequest.Imp {
+
+		var bidderExt adapters.ExtImpBidder
+		var extImpAlkimi openrtb_ext.ExtImpAlkimi
+
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if err := json.Unmarshal(bidderExt.Bidder, &extImpAlkimi); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		var bidFloorPrice floors.Price
+		bidFloorPrice.FloorMinCur = imp.BidFloorCur
+		bidFloorPrice.FloorMin = imp.BidFloor
+
+		if len(bidFloorPrice.FloorMinCur) > 0 && bidFloorPrice.FloorMin > 0 {
+			imp.BidFloor = bidFloorPrice.FloorMin
+		} else {
+			imp.BidFloor = extImpAlkimi.BidFloor
+		}
+		imp.Instl = extImpAlkimi.Instl
+		imp.Exp = extImpAlkimi.Exp
+
+		temp := extObj{AlkimiBidderExt: extImpAlkimi}
+		temp.AlkimiBidderExt.AdUnitCode = imp.ID
+
+		extJson, err := json.Marshal(temp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		imp.Ext = extJson
+		updatedImps = append(updatedImps, imp)
+	}
+
+	return updatedImps, errs
+}
+
+func buildBidderRequest(adapter *adapter, encoded []byte) *adapters.RequestData {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	reqBidder := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     adapter.endpoint,
+		Body:    encoded,
+		Headers: headers,
+	}
+	return reqBidder
+}
+
+// MakeBids will parse the bids from the Alkimi server
+func (adapter *adapter) MakeBids(request *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	var errs []error
+
+	if request == nil || response == nil || adapters.IsResponseStatusCodeNoContent(response) {
+		return nil, nil
+	}
```

Comment:
`MakeRequest` checks for `request == nil` and  `adapters.IsResponseStatusCodeNoContent(response)` checks for `response == nil`. 

Therefore can skip `request == nil || response == nil`

```
if adapters.IsResponseStatusCodeNoContent(response) {
    return nil, nil
}
```

Reply:
Fixed